cmd/wallet-web/src/mixins/gnap/store.js

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+const dbName = 'gnap-store';
+const dbObjectStore = 'gnap-keypair-store';
+
+export async function getGnapKeyPair() {
+  const dbExists = (await window.indexedDB.databases()).map((db) => db.name).includes(dbName);
+
+  if (!dbExists) {
+    return createAndStoreGnapKeyPair();
+  }
+
+  const gnapKeyPair = await fetchStoredGnapKeyPair();
+  return gnapKeyPair;
+}
+
+async function createAndStoreGnapKeyPair() {
+  const gnapKeyPair = await createKeyPair();
+  callOnStore(function (store) {
+    store.add(gnapKeyPair);
+  });
+  return gnapKeyPair;
+}
+
+function fetchStoredGnapKeyPair() {
+  return new Promise(function (resolve) {
+    callOnStore(function (store) {
+      store.get(1).onsuccess = function (event) {
+        return resolve(event.target.result);
+      };
+    });
+  });
+}
+
+export function clearGnapStoreData() {
```

Comment:
follow up: deleting whole database vs clearing data.
Remember: creating database everytime impacts performance.

Reply:
updated only deleted the the key value pair inside the store